### PR TITLE
Added SC module implementation

### PR DESCRIPTION
**Pull request by iulianpascalau**
- added SC module implementation

### CODE REVIEW

File: executors/multiversx/filters/pendingOperationFilter.go [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+		}
+
+		_, err := data.NewAddressFromBech32String(item)
+		if err != nil {
+			return fmt.Errorf("%w on item at index %d", err, index)
+		}
+	}
+
+	return nil
+}
```
Comment (sstanculeanu) [inline executors/multiversx/filters/pendingOperationFilter.go:174]: optional, not a strong opinion, could be reduced to:
```go
func (filter *pendingOperationFilter) checkLists() error {
	err := filter.checkList(filter.allowedEthAddresses, isEthItemValid)
	if err != nil {
		return fmt.Errorf("%w in list AllowedEthAddresses", err)
	}

	err = filter.checkList(filter.deniedEthAddresses, isEthItemValid)
	if err != nil {
		return fmt.Errorf("%w in list DeniedEthAddresses", err)
	}

	err = filter.checkList(filter.allowedMvxAddresses, isMvxItemValid)
	if err != nil {
		return fmt.Errorf("%w in list AllowedMvxAddresses", err)
	}

	err = filter.checkList(filter.deniedMvxAddresses, isMvxItemValid)
	if err != nil {
		return fmt.Errorf("%w in list DeniedMvxAddresses", err)
	}

	return nil
}

func isMvxItemValid(item string) bool {
	_, errNewAddr := data.NewAddressFromBech32String(item)
	return errNewAddr == nil
}

func isEthItemValid(item string) bool {
	return !strings.HasPrefix(item, ethAddressPrefix)
}

func (filter *pendingOperationFilter) checkList(list []string, isItemValid func(item string) bool) error {
	for index, item := range list {
		if item == wildcardString {
			continue
		}

		if !isItemValid(item) {
			return fmt.Errorf("%w on item at index %d", errInvalidItem, index)
		}
	}

	return nil
}
```

File: executors/multiversx/module/testdata/grace.pem [hunk 2]
```diff
@@ -0,0 +1,5 @@
+-----BEGIN PRIVATE KEY for erd1r69gk66fmedhhcg24g2c5kn2f2a5k4kvpr6jfw67dn2lyydd8cfswy6ede-----
+ODY5Yzk4ZGYwYjExZmRhMDgwNGQ3ODM1NGVjYTQzMWY4N2E2ZTkxMGZhZTE0MDgx
+ZjczOGMzMTZhMjVlYTQwOTFlOGE4YjZiNDlkZTViN2JlMTBhYWExNThhNWE2YTRh
+YmI0YjU2Y2MwOGY1MjRiYjVlNmNkNWYyMTFhZDNlMTM=
+-----END PRIVATE KEY for erd1r69gk66fmedhhcg24g2c5kn2f2a5k4kvpr6jfw67dn2lyydd8cfswy6ede-----
```
[the remark this answers is not in the document]
Comment (iulianpascalau) [reply executors/multiversx/module/testdata/grace.pem:5]: done

File: executors/multiversx/module/scCallsModule.go [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+		return nil, err
+	}
+
+	return module, nil
+}
+
+// Close closes any components started
+func (module *scCallsModule) Close() error {
+	errNonceTxsHandler := module.nonceTxsHandler.Close()
+	errPollingHandler := module.pollingHandler.Close()
```
[the remark this answers is not in the document]
Comment (iulianpascalau) [reply executors/multiversx/module/scCallsModule.go:114]: very good point, changed

File: executors/multiversx/filters/pendingOperationFilter.go (hunk 1 above)
Comment (iulianpascalau) [reply executors/multiversx/filters/pendingOperationFilter.go:174]: a little bit reticent on using function pointers, but agree with this solution. Changed

**Review by sstanculeanu — APPROVED** [review]

**Review by dragos-rebegea — APPROVED** [review]

**Pull request merged**